Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py

```diff
@@ -0,0 +1,266 @@
+                    dshield_nattacks:
+                        regex: '^.*\\t.*\\t[0-9]+\\t([0-9]+)'
+                        transform: '\\1'
+                    dshield_name:
+                        regex: '^.*\\t.*\\t[0-9]+\\t[0-9]+\\t([^\\t]+)'
+                        transform: '\\1'
+                    dshield_country:
+                        regex: '^.*\\t.*\\t[0-9]+\\t[0-9]+\\t[^\\t]+\\t([A-Z]+)'
+                        transform: '\\1'
+                    dshield_email:
+                        regex: '^.*\\t.*\\t[0-9]+\\t[0-9]+\\t[^\\t]+\\t[A-Z]+\\t(\\S+)'
+                        transform: '\\1'
+            Example config in YAML where the text in each line until the first
+            whitespace is used as indicator::
+                url: https://ransomwaretracker.abuse.ch/downloads/CW_C2_URLBL.txt
+                ignore_regex: '^#'
+        Args:
+            name (str): node name, should be unique inside the graph
+            chassis (object): parent chassis instance
+            config (dict): node config.
+        """
+        super().__init__(base_url=url, verify=not insecure, proxy=proxy)
+        try:
+            self.polling_timeout = int(polling_timeout)
+        except (ValueError, TypeError):
+            raise ValueError('Please provide an integer value for "Request Timeout"')
+        self.user_agent = user_agent
+        self.encoding = encoding
+        self.feed_name = feed_name
+        if not credentials:
+            credentials = {}
+        self.username = credentials.get('identifier', None)
+        self.password = credentials.get('password', None)
+
+        self.ignore_regex: Optional[Pattern] = None
+        if ignore_regex is not None:
+            self.ignore_regex = re.compile(ignore_regex)
+
+        if indicator is not None:
+            self.indicator = json.loads(indicator)
+            if 'regex' in self.indicator:
+                self.indicator['regex'] = re.compile(self.indicator['regex'])
+            else:
+                raise ValueError(f'{self.feed_name} - indicator stanza should have a regex')
+            if 'transform' not in self.indicator:
+                if self.indicator['regex'].groups > 0:
+                    LOG(f'{self.feed_name} - no transform string for indicator but pattern contains groups')
+                self.indicator['transform'] = r'\g<0>'
+        else:
+            self.indicator = None
+
+        self.fields = json.loads(fields)
+        for f, fattrs in self.fields.items():
+            if 'regex' in fattrs:
+                fattrs['regex'] = re.compile(fattrs['regex'])
+            else:
+                raise ValueError(f'{self.feed_name} - {f} field does not have a regex')
+            if 'transform' not in fattrs:
+                if fattrs['regex'].groups > 0:
+                    LOG(f'{self.feed_name} - no transform string for field {f} but pattern contains groups')
+                fattrs['transform'] = r'\g<0>'
+
+    def build_iterator(self):
```

Comment:
add docstring and explain the meaning of this, not sure that the name is good enough